### PR TITLE
Add Java implementation of Forward Euler

### PR DESCRIPTION
**Pull request by atocil**
Edits are fine, but please provide any feedback you can so I can make my next PRs better

### CODE REVIEW

File: contents/forward_euler_method/code/java/Euler.java [hunk 1]
```diff
@@ -0,0 +1,38 @@
+public class Euler {
+    private static double[] solveEuler(double timestep, int n) {
+        double[] eulerResult = new double[n];
+
+        //Setting the initial condition
+        eulerResult[1] = 1;
+        for(int i = 2; i < eulerResult.length; i++) {
+            eulerResult[i] = eulerResult[i-1] - (3 * eulerResult[i-1] * timestep);
```
Comment (Butt4cak3) [inline contents/forward_euler_method/code/java/Euler.java]: Two things in thise line:

1. The parentheses around `(3 * eulerResult[i-1] * timestep)` aren't needed, although I know why you added them. They make the expression a bit more readable and I think I agree, so I guess we'll keep it.

2. You didn't space out `[i-1]` and again, I understand why. But in this case, I think I like `[i - 1]` more. What do you think?

```suggestion
            eulerResult[i] = eulerResult[i - 1] - (3 * eulerResult[i - 1] * timestep);
```

File: contents/forward_euler_method/code/java/Euler.java [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+        }
+        return eulerResult;
+    }
+
+    private static boolean checkResult(double[] eulerResult, double threshold, double timestep) {
+        boolean isApprox = true;
+
+        for(int i = 1; i < eulerResult.length; i++) {
+            double time = (i - 1) * timestep;
+            double solution = Math.exp(-3*time);
```
Comment (Butt4cak3) [inline contents/forward_euler_method/code/java/Euler.java]: Please space the contents of the parentheses of `Math.exp()` out.

File: contents/forward_euler_method/code/java/Euler.java [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+        for(int i = 2; i < eulerResult.length; i++) {
+            eulerResult[i] = eulerResult[i-1] - (3 * eulerResult[i-1] * timestep);
+        }
+        return eulerResult;
+    }
+
+    private static boolean checkResult(double[] eulerResult, double threshold, double timestep) {
+        boolean isApprox = true;
+
+        for(int i = 1; i < eulerResult.length; i++) {
```
Comment (Butt4cak3) [inline contents/forward_euler_method/code/java/Euler.java]: Same as above. Arrays start at index 0.

**Comment by atocil** [comment]
You're spot on with why my indexing mistakes were there! I also incorporated all of the style changes, let me know if I missed anything or if you notice something else!

File: contents/forward_euler_method/code/java/ForwardEuler.java [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+            eulerResult[i] = eulerResult[i - 1] - (3 * eulerResult[i - 1] * timestep);
+        }
+        return eulerResult;
+    }
+
+    private static boolean checkResult(double[] eulerResult, double threshold, double timestep) {
+        boolean isApprox = true;
+
+        for(int i = 0; i < eulerResult.length; i++) {
+            double time = (i - 1) * timestep;
```
Comment (Butt4cak3) [inline contents/forward_euler_method/code/java/ForwardEuler.java]: I found another error that leads back to the indexing difference between Julia and Java. `time` is just `i * timestep`.

Comment (atocil) [reply contents/forward_euler_method/code/java/ForwardEuler.java]: D'oh

File: contents/forward_euler_method/code/java/ForwardEuler.java [hunk 5]
```diff
@@ -14,7 +14,7 @@ private static boolean checkResult(double[] eulerResult, double threshold, doubl
         boolean isApprox = true;
 
         for(int i = 0; i < eulerResult.length; i++) {
-            double time = (i - 1) * timestep;
+            double time = (i) * timestep;
```
Comment (Butt4cak3) [inline contents/forward_euler_method/code/java/ForwardEuler.java]: Uhm... So, why exactly is `i` in parentheses? =P

Comment (atocil) [reply contents/forward_euler_method/code/java/ForwardEuler.java]: Ha! Because I'm lazy and didn't even look that closely

**Pull request merged**